tests/PINCacheTests/PINCacheTests.m

```diff
@@ -131,6 +172,29 @@ - (void)testObjectRemove
     XCTAssertNil(object, @"object was not removed");
 }
 
+- (void)testObjectRemoveAll
+{
+    NSString *key1 = @"key1";
+    NSString *key2 = @"key2";
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+    
+    [self.cache setObject:key1 forKey:key1];
+    [self.cache setObject:key2 forKey:key2];
+    
+    [self.cache removeAllObjects:^(PINCache *cache) {
+        dispatch_semaphore_signal(semaphore);
+    }];
+    
+    dispatch_semaphore_wait(semaphore, [self timeout]);
+    
+    id object1 = [self.cache objectForKey:key1];
+    id object2 = [self.cache objectForKey:key2];
+    
+    XCTAssertNil(object1, @"not all objects were removed");
+    XCTAssertNil(object2, @"not all objects were removed");
+    XCTAssertTrue(self.cache.memoryCache.totalCost == 0, @"memory cache was not 0 after removing all objects");
```

Comment:
Mind adding this too:
    XCTAssertTrue(self.cache.diskByteCount == 0, @"disk cache byte count was not 0 after removing all objects");